src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -52,6 +55,20 @@ public abstract class SkriptEvent extends Structure {
 
 	public static final Priority PRIORITY = new Priority(600);
 
+	/**
+	 * Checks if the current event is of the given types. Will log an error if false.
+	 * @param events the events to check.
+	 * @return true if the current event is of the given types.
+	 */
+	@SafeVarargs
+	public static boolean isEvent(Class<? extends Event>... events) {
+		ParserInstance parser = ParserInstance.get();
+		if (parser.isCurrentEvent(events))
+			return true;
+		Skript.error("You cannot use this element in " + Utils.a(parser.getCurrentEventName()) + " event.");
```

Comment:
imo this error should be printed by the caller

Reply:
hmm this might make the whole method useless then since it's basically the same as just calling isCurrentEvent with ParserInstance at that point 

this was mainly to simplify syntax that require events but maybe it's not necessary?

Reply:
This was an alternative provided when we switched from `getUsableEvents` to `getUsableStructures`, and it was discussed here:  
https://discord.com/channels/135877399391764480/836220422223036467/1045501306158194738  
However im not opposed to removing it if it doesn't seem useful